bin/generate-phpstorm-stubs-map.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+use PhpParser\Node;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor\NameResolver;
+use PhpParser\NodeVisitorAbstract;
+use PhpParser\ParserFactory;
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$phpStormStubsDirectory = __DIR__ . '/../vendor/jetbrains/phpstorm-stubs';
+
+$phpParser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+
+$fileVisitor = new class() extends NodeVisitorAbstract
+{
+    /** @var string[] */
+    private $classNames = [];
+
+    /** @var string[] */
+    private $functionNames = [];
+
+    public function enterNode(Node $node) : ?int
+    {
+        if ($node instanceof Node\Stmt\ClassLike) {
+            $this->classNames[] = $node->namespacedName->toString();
+
+            return NodeTraverser::DONT_TRAVERSE_CHILDREN;
+        }
+
+        if ($node instanceof Node\Stmt\Function_) {
+            /** @psalm-suppress UndefinedPropertyFetch */
+            $this->functionNames[] = $node->namespacedName->toString();
+
+            return NodeTraverser::DONT_TRAVERSE_CHILDREN;
+        }
+
+        return null;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getClassNames() : array
+    {
+        return $this->classNames;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getFunctionNames() : array
+    {
+        return $this->functionNames;
+    }
+
+    public function clear() : void
+    {
+        $this->classNames    = [];
+        $this->functionNames = [];
+    }
+};
+
+$nodeTraverser = new NodeTraverser();
+$nodeTraverser->addVisitor(new NameResolver());
+$nodeTraverser->addVisitor($fileVisitor);
+
+$map = ['classes' => [], 'functions' => []];
+
+foreach (new DirectoryIterator($phpStormStubsDirectory) as $directoryInfo) {
+    if ($directoryInfo->isDot()) {
+        continue;
+    }
+
+    if (! $directoryInfo->isDir()) {
+        continue;
+    }
+
+    if (in_array($directoryInfo->getBasename(), ['tests', 'meta'], true)) {
+        continue;
+    }
+
+    foreach (new DirectoryIterator($directoryInfo->getPathName()) as $fileInfo) {
+        if ($fileInfo->isDot()) {
+            continue;
+        }
+
+        if (! preg_match('/\.php$/', $fileInfo->getBasename())) {
+            continue;
+        }
+
+        $ast = $phpParser->parse(file_get_contents($fileInfo->getPathname()));
+
+        $nodeTraverser->traverse($ast);
+
+        /** @psalm-suppress UndefinedMethod */
```

Comment:
Seems like `$fileVisitor` has the wrong type at this point?

Reply:
`$fileVisitor` is anonymous class with custom methods :)

Reply:
Ah, indeed :+1: